docs/build-provenance-for-all-registries.md

```diff
@@ -0,0 +1,104 @@
+
+Fulcio will return to you a X.509 signing certificate, with the build provenance properties (including your public key) as OIDs in that certificate. You can now throw away your ephemeral private key.
+
+But we aren’t quite ready to send things to the registry - first you’ll sign your build attestation document (with the Fulcio signing certificate), publish it to Sigstore’s Rekor to get a transparency log entry, and use that to create your Sigstore signature bundle with your signatures over your build artifact, the build attestation document, your transparency log entry, and a timestamp.
+
+You might be wondering how much of this Sigstore interaction you need to implement yourself. Ideally there would be a mature Sigstore library in a language supported by your ecosystem tooling. There are currently libraries for [JavaScript](https://github.com/sigstore/sigstore-js), [Go](https://github.com/sigstore/sigstore-go), [Python](https://github.com/sigstore/sigstore-python), as well libraries under development for [Rust](https://github.com/sigstore/sigstore-rs), [Ruby](https://github.com/sigstore/ruby-sigstore), and [Java](https://github.com/sigstore/sigstore-java). See [sigstore-conformance](https://github.com/sigstore/sigstore-conformance) to ensure client libraries are behaving as expected.
+
+
+# Extending the registry to support build provenance
+
+Now we’re ready to publish the package to the registry, but we need a way to authorize the request. Instead of requiring maintainers to store their password or an unscoped API token in the cloud CI/CD provider, we recommend allowing maintainers to set up a trust relationship between the cloud CI/CD system and your registry via OIDC. For an example of this, see <https://docs.pypi.org/trusted-publishers/>. In fact, registries can implement OIDC trust relationships to authorize publish requests without needing any support from Sigstore client libraries. So this is an excellent scoped project to start with, which is exactly what PyPI did.
+
+Finally, we submit the package and the signed build attestations to the registry. Our work is done… as the CLI tooling! Now we need to verify the provenance on the registry side. See above for a discussion on what Sigstore tooling is available in what programming languages.
+
+For a detailed description of how to verify the Sigstore bundle, see the [Verification section of the Sigstore Client Specification](https://docs.google.com/document/d/1kbhK2qyPPk8SLavHzYSDM8-Ueul9_oxIMVFuWMWKz0E/edit#heading=h.g11ovq2s1jxh). Additionally, the registry will want to verify that the bundle comes from a cloud CI/CD vendor it supports, and (of course) that the hash of the package matches what’s recorded in the Sigstore bundle.
```

Comment:
Why does the registry need to verify that the signature in the bundle is from a supported cloud CI/CD vendor?  Shouldn't that be Fulcio's job?

Reply:
Sigstore/Fulcio is used for several things beyond build provenance, and as a result supports many OIDC providers that don't make sense in this context (like https://github.com/sigstore/fulcio/blob/main/federation/accounts.google.com/config.yaml).